microprofile/WFLY-12590_Integrate_MicroProfile_Fault_Tolerance_2.0.adoc

```diff
@@ -0,0 +1,102 @@
+:keywords:          microprofile,fault-tolerance,openshift
+
+== Overview
+
+Integrate support for Eclipse MicroProfile (MP) Fault Tolerance (FT) for applications deployed in WildFly.
+This proposal pertains to MP FT version 2.0 given the timing of this RFE.
+The integration is based on SmallRye project, specifically `smallrye-fault-tolerance` module.
+The `smallrye-fault-tolerance` module is already in use by projects https://thorntail.io/[Thorntail] (WildFly Swarm) and https://quarkus.io/[Quarkus].
+
+MicroProfile Fault Tolerance specification simplifies the challenges brought by the distributed nature of microservices.
+This is because the communication with external systems is inherently unreliable.
+This increases demand on resiliency of applications.
+
+The patterns brought by this specification are annotations such as `@Timeout`, `@Retry`, `@Fallback`, `@CircuitBreaker`, `@Bulkhead` and `@Asynchronous`.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-12590[WFLY-12590 Integrate MicroProfile Fault Tolerance 2.0]
+* https://issues.jboss.org/browse/EAP7-1347[EAP7-1347 Integrate MP Fault Tolerance 2.0 into EAP]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/WFLY-10522[WFLY-10522 Support Eclipse MicroProfile Config] - dependency; already integrated
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:mnovak@redhat.com[Miroslav Novak]
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+[ ] Engineering
+
+[x] QE
+
+=== Affected Projects or Components
+
+* The specification repository resides at official Eclipse organization on GitHub in https://github.com/eclipse/microprofile-fault-tolerance[eclipse/microprofile-fault-tolerance] repository.
+* The implementation of Eclipse MicroProfile Fault Tolerance will be provided by SmallRye project https://github.com/smallrye/smallrye-fault-tolerance[smallrye/smallrye-fault-tolerance].
+* The underlying library used by _current_ SmallRye implementation is Hystrix which resides in official Netflix GitHub organization https://github.com/Netflix/Hystrix[Netflix/Hystrix].
+
+=== Other Interested Projects
+
+Not interested in our integration per se, but projects that have already integrated MP FT:
+
+* https://thorntail.io/[Thorntail] has integrated MicroProfile Fault Tolerance as a https://github.com/thorntail/thorntail/tree/master/fractions/microprofile/microprofile-fault-tolerance[fraction].
+* https://quarkus.io/[Quarkus] has integrated MicroProfile Fault Tolerance as an https://github.com/quarkusio/quarkus/tree/master/extensions/smallrye-fault-tolerance[extension].
+
+== Requirements
+
+* Support Eclipse MicroProfile Fault Tolerance version 2.0 for all applications deployed in WildFly.
+* [WILL BE CLARIFIED] The extension is either added to all configurations provided by WildFly (standalone and domain mode)
```

Comment:
`WILL BE CLARIFIED` - any estimation when this will be clear ?

Reply:
This will be figured out in the coming weeks. I think Bilge is back this week so this should move quicker now.